diff --git a/a/sponza.mtl b/b/sponza.mtl
index b1fd8c2..7abd7dc 100644
--- a/a/sponza.mtl
+++ b/b/sponza.mtl
@@ -16,6 +16,7 @@ newmtl leaf
 	map_Kd textures\sponza_thorn_diff.tga
 	map_d textures\sponza_thorn_mask.tga
 	map_bump textures\sponza_thorn_ddn.tga
+	map_Ks textures\sponza_thorn_spec.tga
 	bump textures\sponza_thorn_ddn.tga
 
 newmtl vase_round
@@ -31,6 +32,7 @@ newmtl vase_round
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\vase_round.tga
 	map_Kd textures\vase_round.tga
+	map_Ks textures\vase_round_spec.tga
 	map_bump textures\vase_round_ddn.tga
 	bump textures\vase_round_ddn.tga
 
@@ -46,6 +48,7 @@ newmtl Material__57
 	Ks 0.0000 0.0000 0.0000
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\vase_plant.tga
+	map_Ks textures\vase_plant_spec.tga
 	map_Kd textures\vase_plant.tga
 	map_d textures\vase_plant_mask.tga
 
@@ -92,8 +95,9 @@ newmtl bricks
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\spnza_bricks_a_diff.tga
 	map_Kd textures\spnza_bricks_a_diff.tga
-	map_bump textures\spnza_bricks_a_diff.tga
-	bump textures\spnza_bricks_a_diff.tga
+	map_bump textures\spnza_bricks_a_ddn.tga
+    map_Ks textures\spnza_bricks_a_spec.tga
+	bump textures\spnza_bricks_a_ddn.tga
 
 newmtl arch
 	Ns 10.0000
@@ -108,6 +112,7 @@ newmtl arch
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_arch_diff.tga
 	map_Kd textures\sponza_arch_diff.tga
+    map_Ks textures\sponza_arch_spec.tga
 
 newmtl ceiling
 	Ns 10.0000
@@ -137,6 +142,7 @@ newmtl column_a
 	map_Ka textures\sponza_column_a_diff.tga
 	map_Kd textures\sponza_column_a_diff.tga
 	map_bump textures\sponza_column_a_ddn.tga
+    map_Ks textures\sponza_column_a_spec.tga
 	bump textures\sponza_column_a_ddn.tga
 
 newmtl floor
@@ -152,6 +158,8 @@ newmtl floor
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_floor_a_diff.tga
 	map_Kd textures\sponza_floor_a_diff.tga
+    map_Ks textures\sponza_floor_a_spec.tga
+ 
 
 newmtl column_c
 	Ns 10.0000
@@ -168,6 +176,7 @@ newmtl column_c
 	map_Kd textures\sponza_column_c_diff.tga
 	map_bump textures\sponza_column_c_ddn.tga
 	bump textures\sponza_column_c_ddn.tga
+    map_Ks textures\sponza_column_c_spec.tga
 
 newmtl details
 	Ns 10.0000
@@ -182,6 +191,7 @@ newmtl details
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_details_diff.tga
 	map_Kd textures\sponza_details_diff.tga
+    map_Ks textures\sponza_details_spec.tga
 
 newmtl column_b
 	Ns 10.0000
@@ -198,6 +208,7 @@ newmtl column_b
 	map_Kd textures\sponza_column_b_diff.tga
 	map_bump textures\sponza_column_b_ddn.tga
 	bump textures\sponza_column_b_ddn.tga
+    map_Ks textures\sponza_column_b_spec.tga
 
 newmtl Material__47
 	Ns 10.0000
@@ -224,6 +235,8 @@ newmtl flagpole
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_flagpole_diff.tga
 	map_Kd textures\sponza_flagpole_diff.tga
+    map_Ks textures\sponza_flagpole_spec.tga
+    map_Ks textures\sponza_fabric_spec.tga
 
 newmtl fabric_e
 	Ns 10.0000
@@ -238,6 +251,7 @@ newmtl fabric_e
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_fabric_green_diff.tga
 	map_Kd textures\sponza_fabric_green_diff.tga
+    map_Ks textures\sponza_fabric_spec.tga
 
 newmtl fabric_d
 	Ns 10.0000
@@ -252,6 +266,7 @@ newmtl fabric_d
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_fabric_blue_diff.tga
 	map_Kd textures\sponza_fabric_blue_diff.tga
+    map_Ks textures\sponza_fabric_spec.tga
 
 newmtl fabric_a
 	Ns 10.0000
@@ -266,6 +281,7 @@ newmtl fabric_a
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\sponza_fabric_diff.tga
 	map_Kd textures\sponza_fabric_diff.tga
+    map_Ks textures\sponza_fabric_spec.tga
 
 newmtl fabric_g
 	Ns 10.0000
@@ -353,6 +369,7 @@ newmtl vase
 	Ke 0.0000 0.0000 0.0000
 	map_Ka textures\vase_dif.tga
 	map_Kd textures\vase_dif.tga
+    map_bump textures\vase_ddn.tga
 
 newmtl Material__25
 	Ns 10.0000
